actionview/lib/action_view/renderer/streaming_template_renderer.rb

```diff
@@ -43,7 +43,7 @@ def log_error(exception)
     # object that responds to each. This object is initialized with a block
     # that knows how to render the template.
     def render_template(view, template, layout_name = nil, locals = {}) # :nodoc:
-      return [super.body] unless layout_name && template.supports_streaming?
+      return [super.body] unless template.supports_streaming?
```

Comment:
This change (layout_name) was introduced in https://github.com/rails/rails/commit/29078ff8f1561420a405384772c051dc30bdcbf6 but I'm not sure why. It prevents proper streaming for non-layout controller actions. The test appears to assume streaming should work, and so I've removed this condition.